internal/services/namedlocations/named_location_resource.go

```diff
@@ -0,0 +1,125 @@
+			if _, err := uuid.ParseUUID(id); err != nil {
+				return fmt.Errorf("specified ID (%q) is not valid: %s", id, err)
+			}
+			return nil
+		}),
+
+		Schema: map[string]*schema.Schema{
+
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"display_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.NoEmptyStrings,
+			},
+
+			"ip": {
+				Type:         schema.TypeList,
+				ExactlyOneOf: eactlyOneOf,
+				Optional:     true,
+				MaxItems:     1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ip_ranges": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"is_trusted": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"country": {
+				Type:         schema.TypeList,
+				ExactlyOneOf: eactlyOneOf,
+				Optional:     true,
+				MaxItems:     1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"countries_and_regions": {
+							Type:     schema.TypeList,
+							Required: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"include_unknown_countries_and_regions": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func namedLocationResourceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if useMsGraph := meta.(*clients.Client).EnableMsGraphBeta; useMsGraph {
+		return namedLocationResourceCreateMsGraph(ctx, d, meta)
+	}
+	return tf.ErrorDiagF(nil, "azuread_conditional_access_policy does not support AAD Graph. Please enable `EnableMsGraphBeta`")
+}
+
+func namedLocationResourceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if useMsGraph := meta.(*clients.Client).EnableMsGraphBeta; useMsGraph {
+		return namedLocationResourceUpdateMsGraph(ctx, d, meta)
+	}
+	return tf.ErrorDiagF(nil, "azuread_conditional_access_policy does not support AAD Graph. Please enable `EnableMsGraphBeta`")
+}
+
+func namedLocationResourceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if useMsGraph := meta.(*clients.Client).EnableMsGraphBeta; useMsGraph {
+		return namedLocationResourceReadMsGraph(ctx, d, meta)
+	}
+	return tf.ErrorDiagF(nil, "azuread_conditional_access_policy does not support AAD Graph. Please enable `EnableMsGraphBeta`")
+}
+
+func namedLocationResourceDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	if useMsGraph := meta.(*clients.Client).EnableMsGraphBeta; useMsGraph {
+		return namedLocationResourceDeleteMsGraph(ctx, d, meta)
+	}
+	return tf.ErrorDiagF(nil, "azuread_conditional_access_policy does not support AAD Graph. Please enable `EnableMsGraphBeta`")
+}
```

Comment:
In v2, these shim functions will go away as we will no longer support AAD Graph

Reply:
Understood, once there's a PR for removing AAD support I'll refactor all of this to be inline with the new layout